SSA/Core/Tactic.lean

```diff
@@ -67,31 +74,12 @@ macro "simp_peephole" "[" ts: Lean.Parser.Tactic.simpLemma,* "]" "at" ll:ident :
         OpDenote.denote, Expr.op_mk, Expr.args_mk,
         DialectMorphism.mapOp, DialectMorphism.mapTy, List.map, Ctxt.snoc, List.map,
         Function.comp, Valuation.ofPair, Valuation.ofHVector, Function.uncurry,
-        Ctxt.destruct_cons, Ctxt.destruct_nil,
         List.length_singleton, Fin.zero_eta, List.map_eq_map, List.map_cons, List.map_nil,
         bind_assoc, pairBind,
         $ts,*]
 
       try simp (config := {failIfUnchanged := false}) only [Ctxt.Var.toSnoc, Ctxt.Var.last]
-      try generalize $ll { val := 0, property := _ } = a;
-      try generalize $ll { val := 1, property := _ } = b;
-      try generalize $ll { val := 2, property := _ } = c;
-      try generalize $ll { val := 3, property := _ } = d;
-      try generalize $ll { val := 4, property := _ } = e;
-      try generalize $ll { val := 5, property := _ } = f;
-      try simp (config := {failIfUnchanged := false, decide := false, zetaDelta := true}) [TyDenote.toType] at a b c d e f;
-      try clear f;
-      try clear e;
-      try clear d;
-      try clear c;
-      try clear b;
-      try clear a;
-      try revert f;
-      try revert e;
-      try revert d;
-      try revert c;
-      try revert b;
-      try revert a;
+      repeat (generalize_or_fail at $ll)
```

Comment:
This already supports an arbitrary number of generalizations, but I am unsure how to revert an arbitrary number of variables. Anyone has an idea?

Reply:
This now works as I figured out how to revert a variable of a given type.